### PR TITLE
Rewrite Status docs

### PR DESCRIPTION
**Pull request by danielballan**
Includes first draft of release notes for the next release

### CODE REVIEW

**Comment by danielballan** [comment]
Part of the thinking here is to imitate the structure of the [trio API docs](https://trio.readthedocs.io/en/stable/reference-core.html), which alternate:

* narrative content and some illustrative examples (very terse, targeted ones, not complete applied examples like a tutorial would have)
* API docs auto-generated from docstrings

Trio puts all of it on one giant page; I'm going for separate pages because I think it's easier to navigate / less overwhelming. I plan to do the same for Signal and Device, Component, etc. in future PRs.

**Comment by danielballan** [comment]
I agree. Fix pushed.

File: doc/source/api.rst [hunk 1]
```diff
@@ -1,39 +0,0 @@
-
-API
-===
-
-
-
-.. autosummary::
```
Comment (tacaswell) [inline doc/source/api.rst:7]: Are we sure all of these got re-documented someplace else?

File: doc/source/status.rst [hunk 2]
```diff
@@ -1,36 +1,243 @@
...
+Status API details
+------------------
+
+.. autoclass:: ophyd.status.StatusBase
+   :members:
+
+In addition we provide specialized subclasses that know more about the object
 they are tied to.
 
-.. autosummary::
```
Comment (tacaswell) [inline doc/source/status.rst:21]: I like a page-per-class a bit better than having them all inline, but :man_shrugging:

File: doc/source/status.rst [hunk 3]
```diff
@@ -1,36 +1,243 @@
...
+.. code:: python
+
+   from ophyd import Device, Component, SubscriptionStatus
+
+   class MyToyDetector(Device):
+       ...
+       # When set to 1, acquires, and then goes back to 0.
+       acquire = Component(...)
+
+       def trigger(self):
```
Comment (tacaswell) [inline doc/source/status.rst:173]: This does not match what the code expects.   It looks like the code expect this callback to return True/False and then calls `self._finished` for you.

File: doc/source/status.rst (hunk 2 above)
Comment (danielballan) [reply doc/source/status.rst:21]: The way that h11 and trio flow has converted me to having them inline is helpful when the number of classes is small. I would definitely not change the bluesky plans page to take this approach.

File: doc/source/status.rst (hunk 3 above)
Comment (danielballan) [reply doc/source/status.rst:174]: Thanks, fixed.

File: doc/source/api.rst (hunk 1 above)
Comment (danielballan) [reply doc/source/api.rst]: Reverted to leave this as is.

**Pull request merged**